UltrasoundRemoteControl/Widgets/qSlicerAbstractUltrasoundParameterWidget.cxx

```diff
@@ -0,0 +1,270 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through the Applied Cancer Research Unit program of Cancer Care
+  Ontario with funds provided by the Ontario Ministry of Health and Long-Term Care
+
+==============================================================================*/
+
+#include "qSlicerAbstractUltrasoundParameterWidget.h"
+#include "qSlicerAbstractUltrasoundParameterWidget_p.h"
+
+// OpenIGTLinkIO includes
+#include <igtlioCommand.h>
+
+// VTK includes
+#include <vtkXMLDataElement.h>
+#include <vtkXMLUtilities.h>
+
+//-----------------------------------------------------------------------------
+qSlicerAbstractUltrasoundParameterWidgetPrivate::qSlicerAbstractUltrasoundParameterWidgetPrivate(qSlicerAbstractUltrasoundParameterWidget *q)
+  : q_ptr(q)
+  , ParameterName("")
+  , ParameterUnit("") 
+  , CmdSetParameter(igtlioCommandPointer::New())
+  , CmdGetParameter(igtlioCommandPointer::New())
+  , InteractionInProgress(false)
+  , ConnectorNode(NULL)
+{
+}
+
+//-----------------------------------------------------------------------------
+void qSlicerAbstractUltrasoundParameterWidgetPrivate::init()
+{
+  Q_Q(qSlicerAbstractUltrasoundParameterWidget);
+  this->CmdSetParameter->SetName("SetUsParameter");
+  this->CmdSetParameter->BlockingOff();
+  this->CmdSetParameter->SetTimeoutSec(1.0);
```

Comment:
this might be too short (5 sec might be safer; of course, if we receive response sooner then we don't need to wait for the timeout to elapse, we can process the response as soon as we receive it).